anvil/src/eth/api.rs

```diff
@@ -1481,45 +1539,43 @@ impl EthApi {
     /// Retrieves the Anvil node configuration params.
     ///
     /// Handler for RPC call: `anvil_nodeInfo`
-    pub async fn anvil_node_info(&self) -> Result<Value> {
+    pub async fn anvil_node_info(&self) -> Result<NodeInfo> {
         node_info!("anvil_nodeInfo");
 
         let env = self.backend.env().read();
         let fork_config = self.backend.get_fork();
         let tx_order = self.transaction_order.read();
+        let fork_config = match fork_config {
+            Some(fork) => {
+                let config = fork.config.read();
+                NodeForkConfig {
+                    fork_url: Some(config.eth_rpc_url.clone()),
+                    fork_block_number: Some(config.block_number),
+                    fork_retry_backoff: Some(config.backoff.as_millis()),
+                }
+            }
+            None => {
+                NodeForkConfig { fork_url: None, fork_block_number: None, fork_retry_backoff: None }
+            }
+        };
 
-        Ok(json!({
-            // accounts,
-            "current_block_number": self.backend.best_number(),
-            "current_block_timestamp": env.block.timestamp.try_into().unwrap_or(u64::MAX),
-            "current_block_hash": self.backend.best_hash(),
-            "hard_fork": env.cfg.spec_id,
-            "transaction_order": match *tx_order {
-                TransactionOrder::Fifo => "fifo",
-                TransactionOrder::Fees => "fees",
+        Ok(NodeInfo::new(
+            self.backend.best_number(),
+            env.block.timestamp.try_into().unwrap_or(u64::MAX),
+            self.backend.best_hash(),
+            env.cfg.spec_id,
+            match *tx_order {
+                TransactionOrder::Fifo => "fifo".to_owned(),
+                TransactionOrder::Fees => "fees".to_owned(),
```

Comment:
I suggest we use Serialize/Deserialize of `TransactionOrder`(add if missing) and use the type instead of converting to String

Reply:
Makes sense! However though as we're moving the types to the `core` crate we won't have access to `TransactionOrder` as it's on the base crate under the `pool` mod which then imports `core`. What would be the best way to solve this?